code/datums/spells/alien_spells/build_resin_structure.dm

```diff
@@ -0,0 +1,35 @@
+/obj/effect/proc_holder/spell/alien_spell/build_resin
+	name = "Build Resin Structure"
+	desc = "Allows you to create resin structures. Does not work while in space."
+	plasma_cost = 55
+	action_icon_state = "alien_resin"
+
+/obj/effect/proc_holder/spell/alien_spell/build_resin/create_new_targeting()
+	return new /datum/spell_targeting/self
+
+/obj/effect/proc_holder/spell/alien_spell/build_resin/cast(list/targets, mob/living/carbon/user)
+	var/list/resin_buildings = list("Resin Wall (55)" = image(icon = 'icons/obj/smooth_structures/alien/resin_wall.dmi', icon_state = "resin_wall-0"),
```

Comment:
```suggestion
	var/static/list/resin_buildings = list("Resin Wall (55)" = image(icon = 'icons/obj/smooth_structures/alien/resin_wall.dmi', icon_state = "resin_wall-0"),
```
So it ain't made each time you use it and it'll just reuse the one static list.
This won't work if the `show_radial_menu` proc mutates the list/entries though

Reply:
Done